source/patterns/@aws-solutions-constructs/aws-kinesisstreams-gluejob/lib/index.ts

```diff
@@ -0,0 +1,186 @@
+import { Stream, StreamProps } from '@aws-cdk/aws-kinesis';
+import { Bucket } from '@aws-cdk/aws-s3';
+import { Aws, Construct } from '@aws-cdk/core';
+import * as defaults from '@aws-solutions-constructs/core';
+
+export interface KinesisStreamGlueJobProps {
+  /**
+   * Existing instance of Kineses Data Stream. If not set, it will create an instance
+   */
+  readonly existingStreamObj?: Stream;
+  /**
+   * User provided props to override the default props for the Kinesis Stream.
+   *
+   * @default - Default props are used
+   */
+  readonly kinesisStreamProps?: StreamProps | any;
+  /**
+   * @default
+   * User provides props to override the default props for Glue ETL Jobs. This parameter will be ignored if the
+   * existingGlueJob parameter is set
+   *
+   * This parameter is defined as `any` to not enforce passing the Glue Job role which is a mandatory parameter
+   * for CfnJobProps. If a role is not passed, the construct creates one for you and attaches the appropriate
+   * role policies
+   */
+  readonly glueJobProps?: CfnJobProps | any;
+  /**
+   * Existing GlueJob configuration. If this property is provided, any configuration provided through @glueJobProps
+   */
+  readonly existingGlueJob?: CfnJob;
+  /**
+   * @default
+   * A JSON document defining the schema structure of the records in the data stream. An example of such a
+   * definition as below. Either @table or @fieldSchema is mandatory. If @table is provided then @fieldSchema is ignored
+   * 	"FieldSchema": [{
+   *  	"name": "id",
+   *  	"type": "int",
+   *    "comment": "Identifier for the record"
+   *  }, {
+   *    "name": "name",
+   *    "type": "string",
+   *    "comment": "The name of the record"
+   *  }, {
+   *    "name": "type",
+   *    "type": "string",
+   *    "comment": "The type of the record"
+   *  }, {
+   *    "name": "numericvalue",
+   *    "type": "int",
+   *    "comment": "Some value associated with the record"
+   *  },
+   */
+  readonly fieldSchema?: CfnTable.ColumnProperty [];
+  /**
+   * Glue Database for this construct. If not provided the construct will create a new Glue Database.
+   * The database is where the schema for the data in Kinesis Data Streams is stored
+   */
+  readonly database?: CfnDatabase;
+  /**
+   * Glue Table for this construct, If not provided the construct will create a new Table in the
+   * database. This table should define the schema for the records in the Kinesis Data Streams.
+   * Either @table or @fieldSchema is mandatory. If @table is provided then @fieldSchema is ignored
+   */
+  readonly table?: CfnTable;
+  /**
+   * The output data stores where the transformed data should be written. Current supported data stores
+   * include only S3, other potential stores may be added in the future.
+   */
+  readonly outputDataStore?: defaults.SinkDataStoreProps;
+}
+
+export class KinesisStreamGlueJob extends Construct {
```

Comment:
Provide comments above, see any other constructs class for reference

Reply:
construct class attributes have no comments. Checked a few others. Please let me know if you require it. I can add them.